torch/csrc/jit/serialization/source_range_serialization.cpp

```diff
@@ -84,11 +74,20 @@ c10::IValue SourceRangeSerializer::serialize_source(
 
 SourceRangePickler::SourceRangePickler() : srs(new SourceRangeSerializer()) {}
 
-std::vector<char> SourceRangePickler::pickle(const SourceRangeRecords& ranges) {
+std::vector<char> SourceRangePickler::pickle(
+    const SourceRangeRecords& ranges,
+    const SourceRangeTagMap& source_range_tags) {
   std::vector<c10::IValue> ivalues;
   for (const auto& range : ranges) {
+    int64_t source_range_tag{-1};
+    const auto& it = source_range_tags.find(range.range);
+    if (it != source_range_tags.end()) {
```

Comment:
shouldn't we assert that there's always a match? or in what case(s) is it ok to not have a tag?
this shouldn't have to care about BC, right?

Reply:
Yes, that is correct.  Yes makes sense to enforce that. I guess I did that because source range pickler does not control what is in source_range_tag and calling site can call with empty map. But now that you pointed out, it makes sense to enforce stronger restriction to make sure we have generated a valid tag.